src/ray/raylet/worker_pool.cc

```diff
@@ -30,13 +30,25 @@ WorkerPool::~WorkerPool() {
   }
 }
 
-void WorkerPool::StartWorker() {
+uint32_t WorkerPool::Size() const {
+  return static_cast<uint32_t>(actor_pool_.size() + pool_.size());
+}
+
+void WorkerPool::StartWorker(bool force_start) {
```

Comment:
The `force_start` parameter doesn't seem to be used. Can we remove this?

Reply:
it's not used yet, but I think it will be useful soon. The idea here is to let the caller decide if they want to start a worker no matter what (i.e., if we want raylet to pre-start `num_cpus` workers, without considering in-flight worker pool. Otherwise, starting `num_cpus` workers will be serialized.

Reply:
btw, to be clear, the implementation does use this flag.

Reply:
I see. Couldn't we just do the check for `started_worker_pids.empty()` outside of `WorkerPool::Start()` though, through your modified `WorkerPool::Size()`? And in the pre-start case, we can just call `StartWorker` without checking the worker pool size at all.

Reply:
I thought about it and decided against it, because I'd prefer to encapsulate this as an implementation detail and not expose the in-flight worker status through the public interface of the worker pool. I just think it'd be preferable for us to keep thinking about the worker pool as an opaque, self-managing container of workers.

Reply:
I know that this is how we do it in the local_scheduler, but we're in the proper OO land now :) If I had to choose, I'd rather drop the flag than expose the in-flight worker status. The disadvantage of dropping the flag is that all calls to `StartWorker` will serialize starting k workers.

Reply:
Ah, I see. Okay, I'm fine with this. Can you document the `force_start` parameter in the header file though?

Reply:
by the way, with the latest change, the `force_start` is now used in the constructor to speed up starting `num_workers` workers.